File: src/main/java/me/grabsky/azure/commands/AzureCommand.java
package me.grabsky.azure.commands;

import me.grabsky.azure.Azure;
import me.grabsky.azure.configuration.AzureLang;
import me.grabsky.indigo.configuration.Global;
import me.grabsky.indigo.framework.commands.BaseCommand;
import me.grabsky.indigo.framework.commands.ExecutorType;
import me.grabsky.indigo.framework.commands.annotations.DefaultCommand;
import me.grabsky.indigo.framework.commands.annotations.SubCommand;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

import java.util.Collections;
import java.util.List;

public class AzureCommand extends BaseCommand {
    private final Azure instance;

    public AzureCommand(@NotNull Azure instance) {
        super("azure", null, "firedot.command.azure", ExecutorType.ALL);
        this.instance = instance;
    }

    @Override
    public List<String> tabComplete(CommandSender sender, String arg, int index) {
        if (index == 0) return List.of("reload");
        return Collections.emptyList();
    }

    @Override
    public void execute(CommandSender sender, String[] args) {
        if (args.length == 1 && args[0].equalsIgnoreCase("reload")) {
            this.onReload(sender);
            return;
        }
        this.onDefault(sender);
    }

    @DefaultCommand
    public void onDefault(CommandSender sender) {
        AzureLang.send(sender, AzureLang.PLUGIN_HELP);
    }

    @SubCommand
    public void onReload(CommandSender sender) {
        if (sender.hasPermission("firedot.command.azure.reload")) {
            if (instance.reload()) {
                AzureLang.send(sender, Global.RELOAD_SUCCESS);
            } else {
                AzureLang.send(sender, Global.RELOAD_FAIL);
            }
            return;
        }
        AzureLang.send(sender, Global.MISSING_PERMISSIONS);
    }
}

File: src/main/kotlin/me/grabsky/azure/commands/AzureCommand.kt
package me.grabsky.azure.commands

import indigo.libraries.lamp.annotation.Command
import indigo.libraries.lamp.annotation.Default
import indigo.libraries.lamp.annotation.Subcommand
import indigo.libraries.lamp.bukkit.annotation.CommandPermission
import indigo.libraries.lamp.help.CommandHelp
import indigo.plugin.configuration.ServerLocale
import indigo.plugin.extensions.sendMessageOrIgnore
import me.grabsky.azure.Azure
import net.kyori.adventure.text.Component
import org.bukkit.command.CommandSender

@Command("azure")
@CommandPermission("azure.command.azure")
class AzureCommand(private val azure: Azure) {

    @Default
    fun onDefault(sender: CommandSender, commandHelp: CommandHelp<Component>) {
        sender.sendMessageOrIgnore(ServerLocale.COMMAND_USAGE_DIVIDER)
        commandHelp.forEach { sender.sendMessage(it) }
        sender.sendMessageOrIgnore(ServerLocale.COMMAND_USAGE_DIVIDER)
    }

    @Subcommand("reload")
    @CommandPermission("azure.command.azure.reload")
    fun onReload(sender: CommandSender) {
        if (azure.reloadPlugin()) {
            sender.sendMessageOrIgnore(ServerLocale.RELOAD_SUCCES)
            return
        }
        sender.sendMessageOrIgnore(ServerLocale.RELOAD_FAIL)
    }
}